diff --git a/Sources/Paywalls/PaywallColor.swift b/Sources/Paywalls/PaywallColor.swift
index d1e8ff8..085b9ed 100644
--- a/Sources/Paywalls/PaywallColor.swift
+++ b/Sources/Paywalls/PaywallColor.swift
@@ -45,6 +45,12 @@ public struct PaywallColor {
     // Only available from iOS 13
     fileprivate var _underlyingColor: (any Sendable)?
 
+    /// "Designated" initializer
+    private init(stringRepresentation: String, underlyingColor: (any Sendable)?) {
+        self.stringRepresentation = stringRepresentation
+        self._underlyingColor = underlyingColor
+    }
+
 }
 
 // MARK: - Public constructors
@@ -120,11 +126,6 @@ private extension PaywallColor {
 
     #endif
 
-    /// "Designated" initializer
-    private init(stringRepresentation: String, underlyingColor: (any Sendable)?) {
-        self.stringRepresentation = stringRepresentation
-        self._underlyingColor = underlyingColor
-    }
 
 }
 
